Microsoft.Toolkit.Uwp.UI.Controls/WrapPanel/WrapPanel.cs

```diff
@@ -117,16 +117,25 @@ protected override Size MeasureOverride(Size availableSize)
             var parentMeasure = new UvMeasure(Orientation, availableSize.Width, availableSize.Height);
             var spacingMeasure = new UvMeasure(Orientation, HorizontalSpacing, VerticalSpacing);
             var lineMeasure = UvMeasure.Zero;
+            var uIndex = 0;
 
             foreach (var child in Children)
             {
                 child.Measure(availableSize);
-
                 var currentMeasure = new UvMeasure(Orientation, child.DesiredSize.Width, child.DesiredSize.Height);
+                if (currentMeasure.U < 0.1)
+                {
+                    continue; // ignore collapsed items
+                }
 
-                if (parentMeasure.U >= currentMeasure.U + lineMeasure.U + spacingMeasure.U)
+                if (parentMeasure.U >= (uIndex == 0 ? currentMeasure.U : currentMeasure.U + lineMeasure.U + spacingMeasure.U))
```

Comment:
```suggestion
                // If this is the first item, do not add spacing. Spacing is added to the "left"
                double uChange = lineMeasure.U == 0
                    ? currentMeasure.U 
                    : currentMeasure.U + spacingMeasure.U;
                if (parentMeasure.U >=uChange + lineMeasure.U)
                {
                    lineMeasure.U += uChange;
                    lineMeasure.V = Math.Max(lineMeasure.V, currentMeasure.V);
                }
```

Reply:
With the above change there isn't a need to check for an index

Reply:
I've integrated these changes, but I'm not sure what you mean with:
> there isn't a need to [...] accounts for items being collapsed
I mean, removing the index is fine (thanks for that suggestion!), but removing the visibility checks makes the problem from issue #2837 come back again.